### ISSUE TITLE
values are marked deprecated when only type has @deprecated

### ISSUE THREAD

**Issue by rbuckton** [microsoft/TypeScript#39474]
**TypeScript Version:**  4.0.0-dev.20200701

**Search Terms:** jsdoc deprecated tag

**Code**

```ts
interface ESMap<K, V> { }

/** @deprecated */
interface Map<T> extends ESMap<string, T> {}

declare const Map: new <K, V>() => ESMap<K, V>;

new Map<string, number>(); // 'Map' is crossed out
```

**Expected behavior:**

- `Map` the *value* is not deprecated and should not be crossed out.

**Actual behavior:**

- `Map` the *value* is considered deprecated, even though `Map` the *type* is not part of the value declaration of `Map`:

**In VSCode:**
![image](https://user-images.githubusercontent.com/3902892/86840103-2db8e080-c057-11ea-8130-28c30442881a.png "in vscode")

**In Playground:**
![image](https://user-images.githubusercontent.com/3902892/86840849-0e6e8300-c058-11ea-8fb0-8024f29ed6ec.png "in playground")

**Playground Link:** https://www.typescriptlang.org/play/index.html?ts=4.0.0-dev.20200707#code/KYDwDg9gTgLgBAbzgXwNwCh0EsB2NhQBmAhgMbBwCiAygLLFgA8A0gDRwBqAfIipgPQAqQXAACAE2BgowUsXzi4g-tjwES5OPSYAVHqHw5xAZyp0GjYzCi4A5uz2JkmSaQA2xGXFIQcVrQwAXHA4wADucCzs3AAUAJRwALw8NNpRnFwYmOihEWlWNjj2IQCuALYARgRc8ahw-PxwAOTaTXBYpj5+WJIyipLSsvLAijHGwBQAjiVYpADW7TiEEHFAA

**Related Issues:** #38523

**Comment by DanielRosenwasser** [comment]
Probably look into `getSemanticMeaning`, then add some special cases. See if that is useful.

**Comment by sandersn** [comment]
Fixed by #39323
